lib/galaxy_test/api/test_tool_data.py

```diff
@@ -44,6 +44,43 @@ def test_download_field_file(self):
         content = show_field_response.text
         assert content == "This is data 1.", content
 
+    def test_reload(self):
+        show_response = self._get("tool_data/test_fasta_indexes/reload", admin=True)
+        self._assert_status_code_is(show_response, 200)
+        print(show_response.content)
+        data_table = show_response.json()
+        assert data_table["columns"] == ['value', 'dbkey', 'name', 'path']
+
+    def test_show_unknown_raises_404(self):
+        show_response = self._get("tool_data/unknown", admin=True)
+        self._assert_status_code_is(show_response, 404)
+
+    def test_show_unknown_field_raises_404(self):
+        show_response = self._get("tool_data/testalpha/fields/unknown", admin=True)
+        self._assert_status_code_is(show_response, 404)
+
+    def test_reload_unknown_raises_404(self):
+        show_response = self._get("tool_data/unknown/reload", admin=True)
+        self._assert_status_code_is(show_response, 404)
+
+    def test_download_field_unknown_file_raises_404(self):
+        show_field_response = self._get("tool_data/testalpha/fields/data1/files/unknown.txt", admin=True)
+        self._assert_status_code_is(show_field_response, 404)
+
+    def test_delete_without_payload_raises_400(self):
+        delete_response = self._delete("tool_data/testbeta", admin=True)
+        self._assert_status_code_is(delete_response, 400)
+
+    def test_delete_without_values_raises_400(self):
+        payload = {"unknown": "test"}
+        delete_response = self._delete("tool_data/testbeta", data=payload, admin=True)
+        self._assert_status_code_is(delete_response, 400)
+
+    def test_delete_with_wrong_values_raises_400(self):
+        payload = {"values": "wrong"}
+        delete_response = self._delete("tool_data/testbeta", data=payload, admin=True)
+        self._assert_status_code_is(delete_response, 400)
+
     # Following test case rendered invalid by the fix in
     # https://github.com/galaxyproject/galaxy/commit/48f77dc742acf01ddbafafcc4634e69378f1f020#diff-bfb557a99c1f7d646d4968d8d680b885R154.
     # TODO: Restore the test case when test framework allows actions from
```

Comment:
Any chance this can be tested now?
I have tried to uncomment the test and fix it a bit, but ended up with the error message: `Job 1 has no user and no session.` so I assume it is still not possible to run it.
Just to be sure I have tested the `DELETE` endpoint manually and it seems to work, but it would be great if we can test it properly :)

Reply:
I'm working on it, should be possible.

Reply:
There are two fixes in https://github.com/jmchilton/galaxy/pull/new/tool_data_fixes. Want to cherry-pick these into this branch?

Reply:
Done! Thank you so much @jmchilton

Reply:
I will try to create the integration tests for this :)

Reply:
@jmchilton I moved the `test_delete_entry` to an integration test [here](https://github.com/galaxyproject/galaxy/pull/11167/commits/c11eca20d6ddcf6dc97b870280f0b167e1ec903a). Hope that is what you had in mind.

Reply:
Beautiful!